static/settings.ts

```diff
@@ -269,9 +279,10 @@ export class Settings {
             selector: string,
             name: keyof SiteSettings,
             populate: {label: string; desc: string}[],
-            defaultValue: string
+            defaultValue: any,
```

Comment:
I'd like the type here to be `SiteSettings[name]` but I'm not sure if such a thing is possible in ts

Reply:
Using generics it should be possible, but then I don't know if the function would still be usabe in this context. See https://www.typescriptlang.org/play?#code/C4TwDgpgBAyhzAJYDsDmBnKBeKBvAsAFBQlQCGAXFMgK4C2ARhAE4A0RpUDV6wzKqdsVIBjKgwD2EgDYQyyIQF8A3ESKhIsRMAhwEAzDgLCSyMnQhUA1hBASAZrHhI06IZwAmEe2RrTgAGpk0jSWTvquANo2do56LhgAukSKaoT2NMgiSBLI5B4ecLLZEswAPABy5tAQAB46yB6YMQ7hCegAfAAUZhZUVRasUF4+foHBoVTxBpEDEIkAlHhpqYREZAVFECXMXQBEZHtDAIwL65sQxcCl+4dDe7V7C6qEG4WX29e7eyJHUHyhZ7nd5XG4-P60aTSZ5AA

Reply:
Ah thank you! I think this can work